### DISCUSSION TITLE
Lesson6: Error while $ forge create........

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1755] [Q&A] [answered]
This is the code
```
hp@lap:~/solidity/foundry-xxx$ forge create SimpleStorage --rpc-url http://127.0.0.1:8545 --private-k
ey 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```
This is the output
```
[⠒] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:8545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

Can anyone tell me why this error is popping??
NOTE: anvil is working, it's showing all the wallets and the private keys.

**Comment by urosognjenovic** [comment]
Have you tried connecting to Ganache?

Do you get "Listening on 127.0.0.1:8545" at the bottom after running ```anvil``` in the command line?

**Reply by yasharyan0904** [reply]
Yes

**Comment by EngrPips** [comment]
Hello @yasharyan0904, Please use the below command and let us know how it plays out.

```js
forge create src/SimpleStorage.sol:SimpleStorage --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --rpc-url http://127.0.0.1:8545
```
if your `SimpleStorage` contract path is not the same as mine (`src/SimpleStorage.sol:SimpleStorage`)  please use the appropriate path to where your `SimpleStorage` contract is located.

**Reply by yasharyan0904** [reply]
Tried, it's giving the  same error :(

<img width="960" alt="CaptureS" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/144695005/c3fb8744-5ff5-4ece-8faa-30f1230e7898">

**Reply by EngrPips** [reply]
I am convinced you are not running an anvil chain on another terminal.

**Reply by yasharyan0904** [reply]
By writing the following code:
```
forge create src/SimpleStorage.sol:SimpleStorage --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --rpc-url http://127.0.0.1:8545 --force 
```
The error:
```
[⠢] Compiling...
[⠔] Compiling 18 files with 0.8.24
[⠑] Solc 0.8.24 finished in 2.22s
Compiler run successful!
Error: 
error sending request for url (http://127.0.0.1:8545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
- ```- 
The compilation is successful but it is not able to deploy the contract :(

**Reply by EngrPips** [reply]
<img width="1440" alt="Screenshot 2024-05-27 at 17 38 55" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/3337f83b-f0fe-4c98-99c3-cb7aed930ff5">

I re-created your error by killing my terminal just to prove that you are not running an anvil chain on a second terminal.

**Comment by EngrPips** [comment, marked as answer]
<img width="1440" alt="Screenshot 2024-05-27 at 17 38 20" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/fafe7b6f-4c07-414e-8fe6-163329efa64b">

You need to run a second terminal where you will run the `anvil` command to spin up an anvil blockchain, and you leave it running, then you can run  this command `forge create src/SimpleStorage.sol:SimpleStorage --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --rpc-url http://127.0.0.1:8545 ` and everything should work just fine.

**Reply by yasharyan0904** [reply]
EUREKA, it worked . Thanks a lot :)

**Reply by EngrPips** [reply]
Anytime, Friend. Keep crushing it.

**Reply by wilson0922** [reply]
This also helped me out! Thank you guys!

**Reply by EngrPips** [reply]
Awesome, I am glad you unblock yourself. Keep crushing it, Friend.
